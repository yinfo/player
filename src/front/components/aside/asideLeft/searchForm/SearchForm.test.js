// @flow

import React from 'react';
import { shallow, mount } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm component', () => {
  const onSubmitMock = jest.fn();
  const preventDefault = jest.fn();
  const ENTER_KEY = 13;

  it('renders as expected', () => {
    const component = shallow(
      <SearchForm isActive={onSubmitMock} onSearchSubmit={() => {}} />,
    );
    expect(component).toMatchSnapshot();
  });

  it('should sumbit on enterKey press', () => {
    const mockEnterKeyPressEvent = {
      charCode: ENTER_KEY,
      preventDefault,
    };

    const SearchFormComponent = mount(
      <SearchForm onSearchSubmit={onSubmitMock} />,
    );

    SearchFormComponent.find('form').simulate(
      'keypress',
      mockEnterKeyPressEvent,
    );

    expect(onSubmitMock).toHaveBeenCalled();
  });
});
