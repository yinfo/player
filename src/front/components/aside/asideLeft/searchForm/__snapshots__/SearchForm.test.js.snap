// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders as expected 1`] = `ShallowWrapper {}`;
